target/scripts/startup/setup-stack.sh

```diff
@@ -618,13 +618,23 @@ function _setup_saslauthd
 
   # checking env vars and setting defaults
   [[ -z ${SASLAUTHD_MECHANISMS:-} ]] && SASLAUTHD_MECHANISMS=pam
-  [[ ${SASLAUTHD_MECHANISMS:-} == ldap ]] && [[ -z ${SASLAUTHD_LDAP_SEARCH_BASE} ]] && SASLAUTHD_MECHANISMS=pam
-  [[ -z ${SASLAUTHD_LDAP_SERVER} ]] && SASLAUTHD_LDAP_SERVER=localhost
+  [[ -z ${SASLAUTHD_LDAP_SERVER} ]] && SASLAUTHD_LDAP_SERVER="${LDAP_SERVER_HOST}"
   [[ -z ${SASLAUTHD_LDAP_FILTER} ]] && SASLAUTHD_LDAP_FILTER='(&(uniqueIdentifier=%u)(mailEnabled=TRUE))'
 
-  if [[ -z ${SASLAUTHD_LDAP_SSL} ]] || [[ ${SASLAUTHD_LDAP_SSL} -eq 0 ]]
+  [[ -z ${SASLAUTHD_LDAP_BIND_DN} ]] && SASLAUTHD_LDAP_BIND_DN="${LDAP_BIND_DN}"
+  [[ -z ${SASLAUTHD_LDAP_PASSWORD} ]] && SASLAUTHD_LDAP_PASSWORD="${LDAP_BIND_PW}"
+  [[ -z ${SASLAUTHD_LDAP_SEARCH_BASE} ]] && SASLAUTHD_LDAP_SEARCH_BASE="${LDAP_SEARCH_BASE}"
+
+  if [[ "${SASLAUTHD_LDAP_SERVER}" = *'://'* ]]
+  then
+    # %% removes longest matching suffix pattern; # removes shortest matching prefix pattern
+    SASLAUTHD_LDAP_PROTO="${SASLAUTHD_LDAP_SERVER%%://*}://"
+    SASLAUTHD_LDAP_SERVER="${SASLAUTHD_LDAP_SERVER#*://}"
+  elif [[ -z ${SASLAUTHD_LDAP_SSL} ]] || [[ ${SASLAUTHD_LDAP_SSL} -eq 0 ]]
```

Comment:
Why is `SASLAUTH_LDAP_SSL` still used here? Thought you tried to eliminate it?

Reply:
It's still there for better backwards compatibility - shall I add a warning there or shall I remove it completely?

Reply:
Imho as we are about to release v10 - a major release - we warn about breaking changes and that people have to adapt their setup accordingly. As this is such a change i would vote for removing it completely as otherwise we will have the problem at a later point in time.

cc @aendeavor

Reply:
> As this is such a change i would vote for removing it completely as otherwise we will have the problem at a later point in time.

I agree. We can safely remove it.